react/features/app/actions.js

```diff
@@ -124,6 +124,8 @@ function _appNavigateToOptionalLocation(
             // FIXME Turn location's host, hostname, and port properties into
             // setters in order to reduce the risks of inconsistent state.
             location.hostname = defaultLocation.hostname;
+            location.pathname
+                = defaultLocation.pathname + location.pathname.substr(1);
```

Comment:
I'm lazy. Could you tell me why this fixes it? Is where the fake window.location object is created? Then later on something will use the pathname to create the url to redirect to?

Reply:
Sure. I believe this code path was not actually used before (due to other unrelated changes), because `window.location.pathname` was modified directly.

When a bare room is parsed with `parseURIString`, it will not have a host nor a proper pathname. The code in this branch took care of the host (an other associated properties), but not the pathname, which becomes a problem when the root of the deployment is not just the host.